Inc/ST-LIB_HIGH/Protections/Protection.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "C++Utilities/CppUtils.hpp"
+
+#define PROTECTIONTYPE_LENGTH 5
+
+enum ProtectionType : uint64_t {
+    BELOW = std::numeric_limits<uint64_t>::max() - PROTECTIONTYPE_LENGTH + 1,
```

Comment:
Porque no?
```suggestion
    BELOW = UINT64_MAX - PROTECTIONTYPE_LENGTH + 1,
```
Es una macro que especifica la HAL.